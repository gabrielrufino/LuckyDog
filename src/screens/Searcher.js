import React, {useCallback, useEffect, useState} from 'react';
import {Image, StyleSheet, View} from 'react-native';

import Button from '../components/Button';
import DogAPI from '../services/DogAPI';
import Logo from '../assets/images/logo-500px.png';
import Select from '../components/Select';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
  spacer: {
    paddingVertical: 5,
  },
  form: {
    flex: 1,
    justifyContent: 'center',
  },
  logoContainer: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  logo: {
    width: 150,
    height: 150,
  },
});

export default function Searcher({navigation}) {
  const [breeds, setBreeds] = useState([]);
  const [loadingBreeds, setLoadingBreeds] = useState(true);
  const [selectedBreed, setSelectedBreed] = useState('');

  useEffect(() => {
    DogAPI.get('/breeds/list/all')
      .then(({data: {message}}) => {
        setBreeds(Object.keys(message));
      })
      .finally(() => {
        setLoadingBreeds(false);
      });
  }, []);

  const onRiffle = useCallback(
    function () {
      DogAPI.get(`/breed/${selectedBreed}/images/random`).then(
        ({data: {message}}) => {
          navigation.navigate('Dog', {image: message});
        },
      );
    },
    [navigation, selectedBreed],
  );

  const onImFeelingLucky = useCallback(
    function () {
      setSelectedBreed('');

      DogAPI.get('/breeds/image/random').then(({data: {message}}) => {
        navigation.navigate('Dog', {image: message});
      });
    },
    [navigation],
  );

  return (
    <View style={styles.container}>
      <View style={styles.form}>
        <View style={styles.logoContainer}>
          <Image source={Logo} style={styles.logo} />
        </View>
        <View style={styles.spacer} />
        <Select
          options={breeds}
          onSelect={setSelectedBreed}
          selected={selectedBreed}
        />
        <View style={styles.spacer} />
        <Button label="Sortear" onPress={onRiffle} />
        <Button flat label="Estou com sorte" onPress={onImFeelingLucky} />
      </View>
    </View>
  );
}
