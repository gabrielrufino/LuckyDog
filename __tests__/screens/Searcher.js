import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Searcher from '../../src/screens/Searcher';

describe('screens/Searcher.js', () => {
  it('Should renders the search screen correctly', () => {
    const tree = renderer.create(<Searcher />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
