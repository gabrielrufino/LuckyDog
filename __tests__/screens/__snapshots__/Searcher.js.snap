// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/Searcher.js Should renders the search screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingVertical": 5,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#CACAAA",
          "borderRadius": 5,
          "borderWidth": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <RNCPicker
          items={
            Array [
              Object {
                "label": "Selecione um item",
                "textColor": 4288256409,
                "value": undefined,
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingVertical": 5,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#C8AB83",
          "borderRadius": 5,
          "justifyContent": "center",
          "paddingVertical": 14,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontWeight": "bold",
          }
        }
      >
        Sortear
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "justifyContent": "center",
          "paddingVertical": 14,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#C8AB83",
            "fontWeight": "bold",
          }
        }
      >
        Estou com sorte
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#CACAAA",
        "borderRightWidth": 0,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "justifyContent": "center",
        "paddingVertical": 14,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#C8AB83",
          "fontWeight": "bold",
        }
      }
    >
      Favoritos
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "justifyContent": "center",
        "paddingVertical": 14,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#C8AB83",
          "fontWeight": "bold",
        }
      }
    >
      Hist√≥rico
    </Text>
  </View>
</View>
`;
